software/controller/controller.sh

```diff
@@ -116,11 +116,22 @@ install_linux() {
                lcov \
                clang-tidy
   pip3 install -U pip
+  pip3 install nanopb
   pip3 install codecov pyserial matplotlib pandas gitpython
   pip3 install platformio==${PIO_VERSION}
   source ${HOME}/.profile
 }
 
+generate_network_protocols() {
```

Comment:
No, this is not the right way to do it? Why copy paste the same code into different places? This creates additional maintenance burden. One script can call the other.
Or rather, have the expectation of the other script having been run at a prior time.